Gnimuc/ClangCompiler.jl
# Rebase: Misc. updates for LLVM 13/Julia 1.8

- Do not `include` in `__init__()`
- Update bindings
- Add more wrappers for `Decl`s -- Part 1
- Add handles for `Stmt` and `Expr`
- Add wrappers for `VarDecl`
- Add more `Decl` core types
- Add more handles for `ExprCXX` and `StmtCXX`
- Add more wrappers
- Update LibClangEx.jl
- Adjust to the latest changes of libclangex
- Update bindings and apply API updates
- Add wrapper for `convertTypeForMemory`
- Reimplement `get_codegen_module`
- Add `clty_to_llvmty_mem` helper function
- Add sanity check for `pointer_from_objref`
- Clean up `JLLEnvs` module
- Update LibClangEx.jl
- Update Project.toml
- `AbstractCompiler` -> `AbstractClangCompiler`
- Adjust file structure
- Add wrappers for `clang::Interpreter`
